sys/include/net/sock/dtls.h

```diff
@@ -468,6 +479,33 @@
  * return 0;
  * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  *
+ * ### Multi-credential handling
+ *
+ * Each sock needs at least one credential tag to operate. `sock_dtls_create` allows to optionally
+ * assign an initial credential. Extra credentials can be added and removed using
+ * @ref sock_dtls_add_credential and @ref sock_dtls_remove_credential respectively (found in
+ * `net/sock/dtls/creds.h`).
+ *
+ * #### Pre-shared Keys Cipher Suites
+ * In the case of PSK, a server can optionally indicate a hint to help the client to decide which
+ * PSK Identity to use, using @ref sock_dtls_set_psk_identity_hint. The client application can
+ * decide which credential to use based on the sent hint and/or the session information, by
+ * registering a callback with @ref sock_dtls_set_client_psk_cb. If no callback is registered, or
```

Comment:
I would change the order of arguments. The client can register the callback and on execution, it can utilize the hint

Reply:
Sorry, I don't understand, can you suggest the change you imagine here?

Reply:
What I meant was: The client application does not decide which credential to use by **registering** the callback but by **executing** the callback that has been registered **beforehand**. Never mind, reading this again I think it is still understandable this way and shorter.